evalml/objectives/binary_classification_objective.py

```diff
@@ -81,3 +88,25 @@ def validate_inputs(self, y_true, y_predicted):
             raise ValueError("y_true contains more than two unique values")
         if len(np.unique(y_predicted)) > 2 and not self.score_needs_proba:
             raise ValueError("y_predicted contains more than two unique values")
+
+    def _handle_nullable_types(self, y_true):
+        """Transforms y_true to remove any incompatible nullable types according to an objective function's needs.
+
+        Args:
+            y_true (pd.Series): Actual class labels of length [n_samples].
+                May contain nullable types.
+
+        Returns:
+            y_true with any incompatible nullable types downcasted to compatible equivalents.
+        """
+        # Since Objective functions dont have the same safeguards around non woodwork inputs,
+        # we'll choose to avoid the downcasting path since we shouldn't have nullable pandas types
+        # without them being set by Woodwork
+        if isinstance(y_true, pd.Series) and y_true.ww.schema is not None:
```

Comment:
Wanted to highlight this behavior. At least in tests, we often call the objective functions with inputs that don't actually have woodwork types, and the objective functions don't have much in the way of logic to initialize woodwork. Given how late in the pipeline these are used and the fact that the nulalble type or not doesn't have any downstream implications here, I decided to just not initialize Woodwork if the data didn't already have types. No reason to waste the time with type inference when the non woodwork inputs should always be compatible anyway.

Reply:
A user could pass them in if they were using the objective functions directly (I don't think we're at risk of them getting passed in with woodwork types from automl search).

The good news is, we can't get pandas nullable types in numpy inputs, so I think we'd cover user inputs by changing this to pass pandas data without woodwork types to the downcast utils.

Reply:
Allowed pandas series without woodwork types in this commit e5e7a529b